spring-ai-model/src/main/java/org/springframework/ai/model/tool/DefaultToolCallingManager.java

```diff
@@ -217,7 +217,12 @@ private InternalToolExecutionResult executeToolCall(Prompt prompt, AssistantMess
 				.observe(() -> {
 					String toolResult;
 					try {
-						toolResult = toolCallback.call(toolInputArguments, toolContext);
+						if (toolContext != null) {
```

Comment:
This can never be `null`. The `toolContext` cromes from `buildToolContext`, which never returns `null`.

Reply:
You are right. This can never be `null` and I will remove this unnecessary check.